Sources/SourceKitLSP/SourceKitLSPServer.swift

```diff
@@ -2568,91 +2611,60 @@ extension SourceKitLSPServer {
       return index.occurrences(relatedToUSR: related.symbol.usr, roles: .baseOf)
     }
 
-    // TODO: Remove this workaround once https://github.com/swiftlang/swift/issues/75600 is fixed
-    func indexToLSPLocation2(_ location: SymbolLocation) -> Location? {
-      return self.indexToLSPLocation(location)
-    }
-
-    // TODO: Remove this workaround once https://github.com/swiftlang/swift/issues/75600 is fixed
-    func indexToLSPTypeHierarchyItem2(
-      definition: SymbolOccurrence,
-      moduleName: String?,
-      index: CheckedIndex
-    ) -> TypeHierarchyItem? {
-      return self.indexToLSPTypeHierarchyItem(
-        definition: definition,
-        moduleName: moduleName,
-        index: index
-      )
-    }
```

Comment:
It seems that these workarounds might indeed no longer needed when using Swift 6.3 snapshots but the issue does still persist in Swift 6.2. Since SourceKit-LSP still needs to compile using the released Swift 6.2 compiler, we still need them. 

Even if the changes were no longer needed, I’d prefer to remove the workarounds in a separate PR.